code/modules/telesci/gps.dm

```diff
@@ -15,6 +15,7 @@ var/list/SPS_list = list()
 
 /obj/item/device/gps/New()
 	..()
+	if(SPS_list.Find(src)) return //exit if we already did this
```

Comment:
Nonono, this is an awful way to do this.
Make the next three lines into a proc, and then override that proc for the SPS.